kimchi/src/circuits/polynomials/range_check/circuitgates.rs

```diff
@@ -1,95 +1,100 @@
-///     remaining 10 crumbs of v2
-///
-/// Constraints:
-///
-///   For efficiency, the values are constrained differently according to their type.
-///    * 12-bit limbs are constrained with plookups
-///    * 2-bit crumbs are constrained with degree-4 constraints
-///
-/// Layout:
-///
-///  This is how three 88-bit inputs v0, v1 and v2 are layed out and constrained.
-///
-///   * vipj is the jth 12-bit limb of value vi
-///   * vicj is the jth 2-bit crumb limb of value vi
-///
-/// Gate:   RangeCheck0    RangeCheck0    RangeCheck1    Zero
-///    Rows -->
-///         0              1              2              3
-///    0 | v0           | v1           | v2           | 0
-///    1 | copy    v0p0 | copy    v1p0 | crumb   v2c0 | crumb   v2c10 <- MSB
-///    2 | copy    v0p1 | copy    v1p1 | crumb   v2c1 | crumb   v2c11
-///    3 | plookup v0p2 | plookup v1p2 | plookup v2p0 | plookup  v0p0
-///    4 | plookup v0p3 | plookup v1p3 | plookup v2p1 | plookup  v0p1
-///    5 | plookup v0p4 | plookup v1p4 | plookup v2p2 | plookup  v1p0
-///    6 | plookup v0p5 | plookup v1p5 | plookup v2p3 | plookup  v1p1
-///    7 | crumb   v0c0 | crumb   v1c0 | crumb   v2c2 | crumb   v2c12
-///    8 | crumb   v0c1 | crumb   v1c1 | crumb   v2c3 | crumb   v2c13
-///    9 | crumb   v0c2 | crumb   v1c2 | crumb   v2c4 | crumb   v2c14
-///   10 | crumb   v0c3 | crumb   v1c3 | crumb   v2c5 | crumb   v2c15
-///   11 | crumb   v0c4 | crumb   v1c4 | crumb   v2c6 | crumb   v2c16
-///   12 | crumb   v0c5 | crumb   v1c5 | crumb   v2c7 | crumb   v2c17
-///   13 | crumb   v0p6 | copy    v1c6 | crumb   v2c8 | crumb   v2c18
-///   14 | crumb   v0p7 | copy    v1c7 | crumb   v2c9 | crumb   v2c19 <- LSB
-///
-///   The 12-bit chunks are constrained with plookups and the 2-bit crumbs
-///   constrained with degree-4 constraints of the form x*(x - 1)*(x - 2)*(x - 3).
-///
-///   Note that copy denotes a plookup that is deferred to the 4th gate (i.e. Zero).
-///   This is because of the limitation that we have at most 4 lookups per row.
-///   The copies are constrained using the permutation argument.
-///
-/// Gate types:
-///
-///   Different rows are constrained using different CircuitGate types
-///
-///   Row   CircuitGate   Purpose
-///     0   RangeCheck0   Partially constrain v0
-///     1   RangeCheck0   Partially constrain v1
-///     2   RangeCheck1   Fully constrain v2 (and trigger plookups constraints on row 3)
-///     3   Zero          Complete the constraining of v0 and v1 using lookups
-///
-///  Nb. each CircuitGate type corresponds to a unique polynomial and thus
-///       is assigned its own unique powers of alpha
-///```
+//~ Range check circuit gates:
+//~
+//~    The range check gate is comprised of three circuit gates (RangeCheck0, RangeCheck1
+//~    and Zero) and can perform range checks on up to three 88-bit values: v0, v1 and v2.
+//~
+//~    Byte-order:
+//~      * Each value is in little-endian byte order
```

Comment:
Because we want most significant sublimbs on the first few columns of the rows to allow for copies of zeroes to convert the 88-bit checks to 64-bit easily.